File: JolyBell.php
<?php

define('WORK_DIR', file_get_contents("./secrets/work_dir.txt"));
$resp = file_get_contents(file_get_contents("./secrets/vk_api.txt")."-141715625");
$resp = json_decode($resp, true);

foreach($resp['response']['items'] as $item){
  $id = $item['id'];
  $idlast = file_get_contents("JolyBell_last_posted_id.txt");
  if($id == $idlast){
    die("Nothing to do");
  }
  file_put_contents("JolyBell_last_posted_id.txt", $id);

  if($item['marked_as_ads'] == "1"){
    die("Ads");
  }
  if(count($item['copy_history']) > 0){
    die("Repost");
  }
  $title = str_replace(";",",",$item['text']);
  if($title == ""){
    $title = "Jolybell";
  }
  $attachs = $item['attachments'];
  $type = "";
  if($attachs != ""){
    foreach($attachs as $atch){
      switch($atch['type']){
        case 'photo':
          $image = $atch['photo']['sizes'][count($atch['photo']['sizes'])-1]['url'];
          $type = 'img';
          break;

        case 'doc':
          $title .= "\n [Нажмите сюда, чтобы увидеть прикрепеленный к оригиналу GIF](".$atch['doc']['url'].")";
          $type = 'text';
          break;

        case "poll":
          $type = 'poll:'.str_replace("#", "", $atch['poll']['answers'][0]['text']);
          for($j = 1; $j < count($atch['poll']['answers']); $j++){
            $type .= "#".str_replace("#", "", $atch['poll']['answers'][$j]['text']);
          }
          break;
      }
    }
  } else {
    $type = 'text';
  }
  $likes = $item['likes']['count'];
  $reposts = $item['reposts']['count'];
  $comments = $item['comments']['count'];
  $views = $item['views']['count'];
  $lin = $item['from_id']."_".$item['id'];
  file_put_contents("./resources/data/JolyBell.txt", $type.';'.$image.';'.$likes.';'.$reposts.';'.$comments.';'.$views.';'.base64_encode($title).';'.$lin);
  if($type == 'img'){
    $url = $image;
    file_put_contents("./resources/picture/JolyBell.jpg", file_get_contents($url));
  }
  file_put_contents("./logs/JolyBell_python.txt", shell_exec('python3 '.WORK_DIR.'JolyBell.py').PHP_EOL, FILE_APPEND);
}
?>

File: JolyBell.py
import praw
import base64
import re
import sys

work_dir = sys.argv[0].split("JolyBell.py")[0];

client_id_f = open(work_dir+"secrets/reddit_client_id.txt", mode="r", encoding="utf-8")
client_id_s = client_id_f.read();
client_secret_f = open(work_dir+"secrets/reddit_client_secret.txt", mode="r", encoding="utf-8")
client_secret_s = client_secret_f.read();
password_f = open(work_dir+"secrets/reddit_password.txt", mode="r", encoding="utf-8")
password_s = password_f.read();

f = open(work_dir+"resources/data/JolyBell.txt", mode="r", encoding="utf-8")
infs = f.read().split(';', 7);

reddit = praw.Reddit(client_id=client_id_s,
                     client_secret=client_secret_s,
                     password=password_s,
                     user_agent='utidteam.com bot by /u/vitya_schel',
                     username='Jolygolf_bot')

type = infs[0]
originalTitle = base64.b64decode(infs[6]).decode('utf-8');
leaveFullComment = 0
if(type == 'img'):
    if(len(originalTitle) < 300):
        title = originalTitle;
    else:
        title = "Jolybell (текст записи в комментариях)"
        leaveFullComment = 1
    image = work_dir+'resources/picture/JolyBell.jpg'
    submission = reddit.subreddit('jolygolf').submit_image(title, image)
else:
    if(type.split(":")[0] == "poll"):
        title = 'Jolybell'
        submission = reddit.subreddit('jolygolf').submit_poll(title, selftext= originalTitle, options=type.split(":")[1].split("#"), duration=3)
        leaveFullComment = 0;
    else:
        title = 'Jolybell'
        submission = reddit.subreddit('jolygolf').submit(title, selftext=originalTitle)
        leaveFullComment = 0;

likes = str(infs[2]);
reposts = str(infs[3]);
comments = str(infs[4]);
views = str(infs[5]);

source = "JolyBell"
flairtext = "{source} | {l} :l: | {r} :r: | {v} :e:".format(source=source, l=likes, r=reposts, v=views)
if(len(flairtext) > 64):
    flairtext = "{source} |{l}:l: | {r}:r: | {v}:e:".format(source=source, l=likes, r=reposts, v=views)
    if(len(flairtext) > 64):
        source = "JB"
        flairtext = "{source} | {l}:l: | {r}:r: | {v}:e:".format(source=source, l=likes, r=reposts, v=views)

submission.flair.select("2ea55bd8-e3e6-11ea-8a5c-0e7cceef0c57", flairtext)


urlID = infs[7].split("_")[1]
urlFull = "https://jolybot.utidteam.com/away.php?url={urlid}&source=jb".format(urlid=urlID)
if(leaveFullComment == 0):
    comment = submission.reply("[Запись ВКонтакте]({urlFull})".format(urlFull=urlFull))
else:
    infs = originalTitle.splitlines()
    txt = "";
    for line in infs:
        txt += '\n> {l}'.format(l=line)

    txt = re.sub("(> \n)", "", txt);

    comment = submission.reply("[Запись ВКонтакте]({urlFull}) \n\nПолный текст записи: {txt}".format(urlFull=urlFull, txt=txt))
comment.mod.distinguish(how='yes', sticky=True)
comment.mod.approve();
submission.mod.approve();
